README.md

```diff
@@ -0,0 +1,167 @@
+
+## POSTing a Message
+
+Okay, so you got the server to respond: "Hello, World!" Good job! To a large extent, the code the you've just made work represents the main pattern that you'll use through this lab. In each case, you'll define a route, and then within a callback function that you pass in, you will define the "business logic" that the server should follow to produce the desired response. With that in mind, let's move on.
+
+Now that we've built our "Hello World" server, let's start to build out our messaging apparatus. The ability to add messages seems like a good place to start, no? Then we can work on our retrieval pathways.
+
+So what do we need to do? First, of all we need place to store the messages. Let's just store our messages in memory to keep things simple. This will mean that the messages we create won't persist if we shutdown the server, but that's okay for our purposes. To acheive this, let's a define a variable `messages` that will start as an empty list or Array. You can put this somewhere near the top of your `server.js` file:
+
+`let messages = [];`
+
+Note, that in the line above we are using the new ES6 variable assignment keyword `let`. You will need to put the compiler into strict mode for this to work, otherwise you'll likely get an error when you run the code.
+
+Next let's define a new route for submitting messages. In order to send data to the server, we want to use a different HTTP request "method." For the Hello World server we used a standard GET request, but for sending messages we'll use the POST method. A code stub like this will get us started:
+
+```javascript
+route.post('/message', (request, response) => {
+  // Save the message and send the message id back to the client.
+});
+```
+
+At this point, our server can catch a POST request to the path `/message`. Great! But how will get our message data into the system? Now, normally you'd probably have a web frontend that submits the data through the browser. As mentioned above, we'll simulate this by using curl. The following command will submit a message to our new path:
+
+```bash
+curl -X POST http://localhost:3000/message -d '{"id":1,"message":"Our first message."}'
+```
+
+Okay, so this is how we'll submit data, but how will we read in our route's logic? Here's where we will use the `request` object that is passed to ur callback function as an argument. However, although this object contains a lot of information about the request being made to the server, by default it does not contain the data that we've sent across in our curl request in a way that is easily accessed. To make things a bit easier, we'll use another Express module called [body-parser](https://github.com/expressjs/body-parser). As it's name suggests, this module parses data that the server receives from the request. It then attaches this parsed data to the request object's `body` parameter.
```

Comment:
"ur" -> "our"
"it's name" -> "its name"
"server receives from the request" -> (maybe) "server receives from the client" OR "server receives in the request body"